lite2/client.go

```diff
@@ -916,11 +916,25 @@ func (c *Client) compareNewHeaderWithWitnesses(h *types.SignedHeader) error {
 	}
 
 	// 1. Loop through all witnesses.
-	for _, witness := range c.witnesses {
+	for i := 0; i < len(c.witnesses); i++ {
+		witness := c.witnesses[i]
 
 		// 2. Fetch the header.
-		altH, err := witness.SignedHeader(h.Height)
-		if err != nil {
+		altH, err := c.signedHeaderFromWitness(witness, h.Height)
+		switch err.(type) {
+		case nil:
+			break
+		case ErrWitnessNonResponsive: // remove witness
```

Comment:
not clear to me why we need this block? if witness is not responsive, we should just remove it and continue with the rest of the witnesses

Reply:
Do we need to make the distinction between errors or regardless of the errors should we then remove it and continue

Reply:
regardless